int32validator/all_example_test.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package int32validator_test
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int32validator"
+)
+
+func ExampleAll() {
+	// Used within a Schema method of a DataSource, Provider, or Resource
+	_ = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"example_attr": schema.Int32Attribute{
+				Required: true,
+				Validators: []validator.Int32{
+					// Validate this Int32 value must either be:
+					//  - 1.0
+					//  - At least 2.0, but not 3.0
+					int32validator.Any(
+						int32validator.OneOf(1.0),
+						int32validator.All(
+							int32validator.AtLeast(2.0),
+							int32validator.NoneOf(3.0),
+						),
+					),
```

Comment:
nit: I see the Int64 examples also have this, but probably should use a value more likely to be seen here, `1`, `2`, `3`, etc.
```suggestion
					// Validate this Int32 value must either be:
					//  - 1
					//  - At least 2, but not 3
					int32validator.Any(
						int32validator.OneOf(1),
						int32validator.All(
							int32validator.AtLeast(2),
							int32validator.NoneOf(3),
						),
					),
```